src/nft/queries/genie/SweepFetcher.ts

```diff
@@ -24,22 +36,37 @@ export const fetchSweep = async ({
       paymentToken: 1,
       animationUrl: 1,
       notForSale: 1,
+      rarity: 1,
+      tokenId: 1,
     },
-    limit: 99,
+    limit: 50,
     offset: 0,
   }
 
   if (markets) {
     payload.markets = markets
   }
-
+  const numberOfTraits = traits?.filter((trait) => trait.trait_type === 'Number of traits')
+  if (numberOfTraits) {
+    payload.filters.numTraits = numberOfTraits.map((el) => ({ traitCount: el.trait_value }))
+  }
```

Comment:
If you need to refactor off Alex's PR I believe we remove `number of traits` and this logic can be removed

Reply:
W